### DISCUSSION TITLE
Make it easier for people to contribute

### DISCUSSION THREAD

**Discussion by paulcsmith** [luckyframework/lucky#593] [General]
Inspired by @robacarp's post: 
https://gitter.im/luckyframework/Lobby?at=5b8751609c71d363c14b2bba

### Links

* https://solnic.codes/2018/08/29/a-quick-recap-and-plans-for-the-future/
* http://jessenoller.com/blog/2015/9/27/a-lot-happens

### Goals

* Make it easier to share the workload
* Make it easier for people to get started in open source
* Make it clear why some contributions won't be accepted, or why it will take longer to review
* Automate common steps (like new releases) so it is easier for myself or anyone else to keep things moving forward
* Help build a community that communicates clearly, humbly, and respectfully
* Help people feel comfortable making suggestions, and responding to suggestions kindly.

### For all Lucky libraries

- [x] Add `scripts/setup` scripts that make it easy to get started
- [x] Add `scripts/test` scripts that run any necessary tests (`crystal spec`, `shards build`, check formatting)
- [x] Start asking for inline docs. Add `PULL_REQUEST.md` template that guides users to write docs, run `bin/test`, check formatting.
- [ ] Add `bin/prep_release` scripts to all Lucky libraries. This would make it easier to bump versions by updating shard.yml, version.cr, anything in the README and then tag it with the correct version number
- [x] Me: Start adding more details when opening PRs and issue that talk about *why* I'm doing things

### Improve the trickier parts of Lucky

- [ ] Lucky CLI templates are very confusing right now. This may required improving teeplate to support conditional files/folders. Right now we have folders all over because that is not supported. It makes it hard to tell what a project will actually look like
- [x] LuckyRecord see if we can remove or extract some of the macros to nicely named methods
- [ ] LuckyRecord form code is pretty complex in some spots. Needs to be simplified

 More to come

### Not sure where...

- [ ] Talk about why certain contributions take longer to review
- [ ] Talk about why certain features (Windows support, MySQL/sqlite support) are not done and likely won't be done for awhile

More to come :)

**Comment by ajwann** [comment]
@paulcsmith I would like to create the ```test``` scripts. Is a bin directory with an executable test file that runs those three commands all that you are looking for? Is it safe to assume that all lucky libraries would need to have those three commands run?

**Comment by paulcsmith** [comment]
@ajwann that would be wonderful. I don’t think all of them need to run shards build but it wouldn’t hurt. 

And yes an executable that does that would be great!

**Comment by ajwann** [comment]
@paulcsmith sounds good, I'll have a PR up shortly.

**Comment by 5minpause** [comment]
@paulcsmith 
> Add `bin/setup` scripts that make it easy to get started

Can you elaborate what is missing for this currently?
